staging/operator-lifecycle-manager/pkg/controller/registry/reconciler/grpc.go

```diff
@@ -186,17 +186,34 @@ func (c *GrpcRegistryReconciler) currentPodsWithCorrectImage(source grpcCatalogS
 	return found
 }
 
+func (c *GrpcRegistryReconciler) currentPodsWithCorrectHash(source grpcCatalogSourceDecorator, saName string) []*corev1.Pod {
+	pods, err := c.Lister.CoreV1().PodLister().Pods(source.GetNamespace()).List(labels.SelectorFromValidatedSet(source.Labels()))
+	if err != nil {
+		logrus.WithError(err).Warn("couldn't find pod in cache")
+		return nil
+	}
+	found := []*corev1.Pod{}
+	newPod := source.Pod(saName)
+	for _, p := range pods {
+		if PodHashMatch(p, newPod) {
+			found = append(found, p)
+		}
+	}
+	return found
+}
```

Comment:
Unfortunately, and it's possible I'm missing something here, we are trying to avoid touching any code in the staging tree that hasn't been introduced/modified/etc. upstream. This is to ensure that the staging directory is 1:1 with the upstream repositories. We may need to pivot to a more generalized solution for injecting annotations into the catalogsource(s).

Reply:
I can try to convert this solution to something more upstream-friendly. However, one of the reasons we split downstream and upstream is the possibility of having downstream-only changes. This is potentially one of the cases. Where do you say I should make those changes then?